rust/geoarrow-array/src/cast.rs

```diff
@@ -820,6 +820,950 @@ macro_rules! downcast_geoarrow_array {
     };
 }
 
+/// Two argument version of the `downcast_geoarrow_array!` macro.
+/// Downcast any combination of two [GeoArrowArray] to a concrete-typed array based on its [`GeoArrowType`].
+#[macro_export]
+macro_rules! downcast_geoarrow_array_two_args {
```

Comment:
Can we temporarily put this in the `geoarrow-geo` crate and remove the `#[macro_export]`? I like this approach so that we can iterate on useful code while not adding to the public API and thus not forcing ourselves into future backwards compatibility.